sdk/testing/src/main/java/io/opentelemetry/sdk/testing/assertj/LongGaugeAssert.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.testing.assertj;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.opentelemetry.sdk.metrics.data.GaugeData;
+import io.opentelemetry.sdk.metrics.data.LongPointData;
+import java.util.Arrays;
+import java.util.function.Consumer;
+import javax.annotation.Nullable;
+import org.assertj.core.api.AbstractAssert;
+
+/** Assertions for an exported long {@link GaugeData}. */
+public final class LongGaugeAssert
+    extends AbstractAssert<LongGaugeAssert, GaugeData<LongPointData>> {
+
+  LongGaugeAssert(@Nullable GaugeData<LongPointData> actual) {
+    super(actual, LongGaugeAssert.class);
+  }
+
+  /**
+   * Asserts the gauge has points matching all of the given assertions and no more, in any order.
+   */
+  @SafeVarargs
+  @SuppressWarnings("varargs")
+  public final LongGaugeAssert hasPointsSatisfying(Consumer<LongPointDataAssert>... assertions) {
+    return hasPointsSatisfying(Arrays.asList(assertions));
+  }
+
+  /**
+   * Asserts the gauge has points matching all of the given assertions and no more, in any order.
+   */
+  public LongGaugeAssert hasPointsSatisfying(
```

Comment:
I had originally combined long/double versions of assertions into one class since they were equivalent, but I wasn't able to get generic-fu to work with this sort of method so went with splitting again.